### PR TITLE
Implement adder for model attribute casting

### PR DESCRIPTION
**Pull request by scottbedard**
Closes https://github.com/octobercms/october/issues/3407

### CODE REVIEW

File: src/Database/Model.php [hunk 1]
```diff
@@ -672,6 +672,17 @@ protected function performDeleteOnRelations()
     //
     // Adders
     //
+    
+    /**
+     * Add attribute casts for the model.
+     * 
+     * @param  array              $attributes
```
Comment (LukeTowers) [inline src/Database/Model.php]: What's with all the space between array and attributes?

Comment (scottbedard) [reply src/Database/Model.php]: Heh, couldn't tell you... I guess that's just what my editor decided to do. I'll go ahead and trim that.

**Comment by LukeTowers** [comment]
Could you add some information to the docs as well for this and the other protected setters if there isn't any already?

**Comment by scottbedard** [comment]
@LukeTowers I couldn't find any, so I've added a blurb to the end of our model docs.

File: src/Database/Model.php [hunk 2]
```diff
@@ -672,6 +672,17 @@ protected function performDeleteOnRelations()
...
     // Adders
     //
+    
+    /**
+     * Add attribute casts for the model.
+     * 
+     * @param  array $attributes
+     * @return void
+     */
+    public function addCasts($attributes)
```
Comment (LukeTowers) [inline src/Database/Model.php]: Could you make this match the other setters? I.e. pass multiple strings or an array of strings?

Comment (scottbedard) [reply src/Database/Model.php]: No, that API doesn't make sense in this context. Unlike the other adders, a model's `casts` value is an associative array.

Comment (LukeTowers) [reply src/Database/Model.php]: Fair enough

**Comment by scottbedard** [comment]
Woo! Thanks @LukeTowers for getting this merged so quickly!

**Comment by LukeTowers** [comment]
Thanks for the contribution @scottbedard!

**Pull request merged**